kornia/contrib/face_detection.py

```diff
@@ -0,0 +1,411 @@
+# based on: https://github.com/ShiqiYu/libfacedetection.train/blob/74f3aa77c63234dd954d21286e9a60703b8d0868/tasks/task1/yufacedetectnet.py  # noqa
+from enum import Enum
+from itertools import product
+from typing import cast, List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from kornia.geometry.bbox import nms
+
+__all__ = [
+    "FaceDetector",
+    "FaceDetectorResults",
+    "FaceKeypoint",
+]
+
+
+url: str = "https://github.com/ShiqiYu/libfacedetection.train/raw/master/tasks/task1/weights/yunet_final.pth"
+
+
+class FaceKeypoint(Enum):
+    r"""Define the keypoints detected in a face.
+
+    The left/right convention is based on the screen viewer.
+    """
+    EYE_LEFT = 0
+    EYE_RIGHT = 1
+    NOSE = 2
+    MOUTH_LEFT = 3
+    MOUTH_RIGHT = 4
+
+
+class FaceDetectorResults:
```

Comment:
Agree, we should keep things consistent

Reply:
Ideally we should enable the trainer to easily fine tune the network — and explore if this light network could be used for other detection problems. Not sure about that one since it relies on the facial keypoints but still worth trying.